File: java/src/test/java/com/dddeurope/niuz/services/TeaserServiceTest.java
package com.dddeurope.niuz.services;

import com.dddeurope.niuz.dtos.TeaserDto;
import com.dddeurope.niuz.entities.Teaser;
import com.dddeurope.niuz.repositories.TeaserRepository;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class TeaserServiceTest {
    private TeaserRepository teasers;
    private TeaserService teaserService;

    @BeforeEach
    void setUp() {
        teasers = mock(TeaserRepository.class);
        teaserService = new TeaserService(teasers);
    }

    @Test
    void getHomepage() {
        when(teasers.getByPage("homepage")).thenReturn(Arrays.asList(new Teaser("headline", "Freddy Kruger")));
        assertThat(teaserService.get("homepage")).containsExactly(
                new TeaserDto("headline", "Freddy Kruger")
        );
    }
}

File: dotnet/niuz.application.test/services/TeaserServiceTest.cs
using System.Collections.Generic;
using FluentAssertions;
using niuz.application.dtos;
using niuz.application.entities;
using niuz.application.repositories;
using NSubstitute;
using Xunit;

namespace niuz.application.services
{
  public class TeaserServiceTest : BaseTest
  {
    [Fact]
    public void GetHomepage()
    {
      var teaserService = Get<TeaserService>();

      teasers.GetByPage("homepage").Returns(new List<Teaser> {new Teaser("headline", "Freddy Kruger")});

      teaserService.Get("homepage").Should().ContainInOrder(
        new TeaserDto("headline", "Freddy Kruger")
      );

    }
  }
}
